apps/src/turtle/customLevelBlocks.js

```diff
@@ -39,28 +39,80 @@ function createACircleCode (size, gensym, indent) {
     indent + '}\n'].join('\n');
 }
 
-
 /**
- * Same as draw_a_square, except inputs are not inlined
+ * Returns an initialization object that sets up blockly attributes;
+ *
+ * @param title - The title of the block that will be visible to the user.
+ * @param [argument] - Optional parameter for blocks that accept a value
+ *    argument. This is the title of the argument.
+ * @param [titleValue] - Optional parameter to associate a value with the
+ *    title.
+ * @return the initialization object
  */
-function installDrawASquare(blockly, generator, gensym) {
-  // Create a fake "draw a square" function so it can be made available to users
-  // without being shown in the workspace.
-  blockly.Blocks.draw_a_square_custom = {
-    // Draw a square.
+function makeInitializer(title, argument, titleValue) {
+  return {
     init: function() {
       this.setHSV(94, 0.84, 0.60);
-      this.appendDummyInput()
-          .appendTitle(msg.drawASquare());
-      this.appendValueInput('VALUE')
-          .setAlign(blockly.ALIGN_RIGHT)
-          .setCheck(Blockly.BlockValueType.NUMBER)
-              .appendTitle(msg.lengthParameter() + ':');
+
+      this.appendDummyInput().appendTitle(title, titleValue);
+
+      if (argument !== undefined) {
+        this.appendValueInput('VALUE')
+            .setAlign(Blockly.ALIGN_RIGHT)
+            .setCheck(Blockly.BlockValueType.NUMBER)
+            .appendTitle(argument + ':');
+      }
+
       this.setPreviousStatement(true);
       this.setNextStatement(true);
       this.setTooltip('');
     }
   };
+}
+
+/**
+ * Returns an initialization object that sets up blockly attributes on blocks
+ * that associate a value with their titles.
+ *
+ * @param title - The title of the block that will be visible to the user.
+ * @param value - The value to associate with the title.
+ * @return the initialization object
+ */
+function makeInitializerWithAssociatedTitleValue(title, value) {
+  return makeInitializer(title, undefined, value);
+}
+
+/**
+ * Returns an initialization object that sets up blockly attributes on blocks
+ * that take a size parameter.
+ *
+ * @param title - The title of the block that will be visible to the user.
+ * @return the initialization object
+ */
+function makeInitializerWithSizeParam(title) {
+  return makeInitializer(title, msg.sizeParameter());
+}
```

Comment:
These first two helpers appear to only get used in one spot each. Any reason not just just have callers use makeInitializer directly?